src/api.ts

```diff
@@ -3855,6 +3856,159 @@ export const methods = {
+          latestVersion: archiver.latestVersion || current.latestVersion,
+          minVersion: archiver.minVersion || current.minVersion,
+          certCycleDuration: current.certCycleDuration,
+          maintenanceFee: current.maintenanceFee,
+          maintenanceInterval: current.maintenanceInterval,
+          penalty: {
+            amount: current.nodePenaltyUsd.value,
+            currency: 'shm',
+          },
+          slashing: {
+            leftNetworkEarlyPenaltyPercent: current.slashing.leftNetworkEarlyPenaltyPercent,
+            nodeRefutedPenaltyPercent: current.slashing.nodeRefutedPenaltyPercent,
+            syncTimeoutPenaltyPercent: current.slashing.syncTimeoutPenaltyPercent,
+          },
+          reward: {
+            amount: current.nodeRewardAmountUsd.value,
+            currency: 'shm',
+            nodeRewardInterval: current.nodeRewardInterval,
+          },
+          requiredStake: {
+            amount: current.stakeRequiredUsd.value,
+            currency: 'shm',
+          },
+          restakeCooldown: current.restakeCooldown,
+          timestamp: networkAccountData.networkAccount.timestamp,
+        }
+
+        logEventEmitter.emit('fn_end', ticket, { success: true }, performance.now())
+        callback(null, response)
+        countSuccessResponse(api_name, 'success', 'archiver')
+      } else {
+        logEventEmitter.emit('fn_end', ticket, { success: false }, performance.now())
+        callback({ code: -32000, message: 'Network account data not found or invalid' }, null)
+        countFailedResponse(api_name, 'Network account data not found or invalid')
+      }
+    } catch (error: any) {
+      console.error('Error fetching network account data:', error)
+      logEventEmitter.emit('fn_end', ticket, { success: false }, performance.now())
+      callback({ code: -32000, message: 'Failed to fetch network account data', data: error.message }, null)
+      countFailedResponse(api_name, 'Failed to fetch network account data')
+    }
+  },
+  shardeum_getCycleInfo: async function (args: RequestParamsLike, callback: JSONRPCCallbackTypePlain) {
+    const api_name = 'shardeum_getCycleInfo'
+    nestedCountersInstance.countEvent('endpoint', api_name)
+
+    if (
+      !Array.isArray(args) ||
+      args.length > 1 ||
+      (args.length === 1 && typeof args[0] !== 'number' && args[0] !== null)
+    ) {
+      callback({ code: -32602, message: 'Invalid params: Expected a single number or null.' }, null)
+      countFailedResponse(api_name, 'Invalid params: Expected a single number or null.')
+      return
+    }
+    const ticket = crypto
+      .createHash('sha1')
+      .update(api_name + Math.random() + Date.now())
+      .digest('hex')
+
+    logEventEmitter.emit('fn_start', ticket, api_name, performance.now())
+    const cycleNumber = args.length === 1 ? args[0] : null
+    try {
+      const result = await collectorAPI.getCycleInfo(cycleNumber)
```

Comment:
Please add proper `.then()` and `.catch()` here to ensure we handles errors coming from the cycleInfo endpoint. The `then()` should handle empty response from the collector, and the catch() should ensure errors are handled and `result` is only given a value in an ideal case, and `undefined` otherwise.